Plugin/Framework/Search/Request/GraphQlHandler.php

```diff
@@ -0,0 +1,118 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Plugin\Framework\Search\Request;
+
+use Magento\Store\Model\StoreManagerInterface;
+use Nosto\Cmp\Helper\SearchEngine;
+use Nosto\Cmp\Logger\LoggerInterface;
+use Nosto\Cmp\Model\Filter\GraphQlFilters;
+use Nosto\Cmp\Model\Service\Recommendation\StateAwareCategoryServiceInterface;
+use Nosto\Cmp\Plugin\Catalog\Block\ParameterResolverInterface;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+
+class GraphQlHandler extends AbstractHandler
+{
+
+    /** @var GraphQlFilters */
+    private $filters;
+
+    /**
+     * GraphQlHandler constructor.
+     * @param GraphQlFilters $filters
+     * @param ParameterResolverInterface $parameterResolver
+     * @param SearchEngine $searchEngineHelper
+     * @param StoreManagerInterface $storeManager
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param StateAwareCategoryServiceInterface $categoryService
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        GraphQlFilters $filters,
+        ParameterResolverInterface $parameterResolver,
+        SearchEngine $searchEngineHelper,
+        StoreManagerInterface $storeManager,
+        NostoHelperAccount $nostoHelperAccount,
+        StateAwareCategoryServiceInterface $categoryService,
+        LoggerInterface $logger
+    ) {
+        parent::__construct(
+            $parameterResolver,
+            $searchEngineHelper,
+            $storeManager,
+            $nostoHelperAccount,
+            $categoryService,
+            $logger
+        );
+        $this->filters = $filters;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getBindKey()
+    {
+        return self::KEY_BIND_TO_GRAPHQL;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function preFetchOps(array $requestData)
+    {
+        $this->categoryService->setCategoryInRegistry(
+            $requestData[self::KEY_FILTERS][self::KEY_CATEGORY_FILTER][self::KEY_VALUE]
+        );
+        $this->filters->setRequestData($requestData);
+    }
+
+    /**
+     * @param array $requestData
+     * @return int
+     */
+    public function parseLimit(array $requestData)
+    {
+        return (int) $requestData[self::KEY_RESULT_SIZE];
```

Comment:
This doesn't hold the real page size used in a query. This seems to be always 1000 (most likely configurable). We need a way to figure out the real page size. One option is to create another plugin for class  `Magento\CatalogGraphQl\Model\Resolver\Products\Query` and store the real page size.